python/paddle/fluid/optimizer.py

```diff
@@ -13,21 +13,23 @@
 # limitations under the License.
 
 from __future__ import print_function
-import re
-import sys
+
 from collections import defaultdict
+from contextlib import contextmanager
+
 from paddle.fluid.framework import Program, Variable, name_scope, default_main_program
+from paddle.fluid.transpiler.details.distribute_lookuptable_utils import find_distributed_lookup_table
```

Comment:
I think `optimizer.py` should not depend on the transpiler and nothing should. And we'll move it to IR pass later, this may cause the rewrite a bit harder.

Reply:
find_distributed_lookup_table is only a pure function, has no relationship with transpiler.

Reply:
So just put it in `optimizer.py`

Reply:
This function is also used in distribute transpiler.

Reply:
done